Sources/MapboxMaps/Gestures/GestureManager+GestureHandlerDelegate.swift

```diff
@@ -86,6 +86,20 @@ extension GestureManager: GestureHandlerDelegate {
         mapView.mapboxMap.setCamera(to: CameraOptions(anchor: anchor, zoom: newScale))
     }
 
+    internal func pinchCenterChanged(offset: CGSize) {
+        guard let mapView = cameraManager.mapView else {
+            Log.error(forMessage: "MapView must exist when the zoom level changes", category: "Gestures")
+            return
+        }
+
+        let currentCenterCoordinate = mapView.mapboxMap.cameraState.center
+        let currentCenterScreenPoint = mapView.mapboxMap.point(for: currentCenterCoordinate)
+        let newCenterScreenPoint = currentCenterScreenPoint.shifted(xOffset: -offset.width,
+                                                                    yOffset: -offset.height)
+        let newCenterCoordinate = mapView.mapboxMap.coordinate(for: newCenterScreenPoint)
+        mapView.mapboxMap.setCamera(to: CameraOptions(center: newCenterCoordinate))
+    }
```

Comment:
Instead of using a setCamera, I'd like to see us use the Drag API for all panning on the map. @cmilack Let me know if you need help with this. But there should be a function called drag that you can use

Reply:
I updated this to use the drag API but it's creating some awkward behavior. It seems to over-compensate when panning which causes the map to shift much more than expected.  I've tried a variety of things to get this right and the implementation I had (w/o drag API) seems to work the best.  From what I've seen, I think that the DragAPI is just not behaving nicely when we're zooming simultaneously. I'm happy to try to dig deeper but not sure how much time/effort I should be putting in here.


1. Two setCamera calls (panning w/o drag API):  works as expected w/ smooth panning and scrolling

2. Two setCamera calls (panning w/ drag API): map pans more than expected and gets worse as touch points are further apart.

3. Single Camera update using dragOptions (drag API) and updating zoom:  similar result to example 2 but with more jitter